src/main/java/com/softteam/vocabuilder/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.softteam.vocabuilder.service;
+
+import com.softteam.vocabuilder.exections.CategoryNotFoundException;
+import com.softteam.vocabuilder.persistence.entity.Category;
+import com.softteam.vocabuilder.persistence.repository.CategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class CategoryServiceImpl implements ICategoryService{
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Override
+    public Category create(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    @Override
+    public void update(Category category) {
+        Optional<Category> category1 = categoryRepository.findById(category.getId());
+        if(category1.isEmpty()){
+            throw new CategoryNotFoundException("category not found",HttpStatus.NOT_FOUND);
+        }
+        categoryRepository.save(category);
```

Comment:
en casi todos los metodos estas yendo a la base de datos dos veces, una para buscar el elemento, y otra para hacer la operacion del metodo, esto no es eficiente. La mayoria de esos metodos te van a retornar algo, ya sea el elemento creado, modificado, algun indicador de que se elimino, etc. Con esa respuesta puedes deducir si estaba el elemento o no, o si dio algun error al hacer la operacion

Reply:
I tried with this but i don't want to change everything it steals this one works fine.
I tried it with this code so far it seems to work but I don't want to change everything with this enhancement until it works fine.
So I need a hand.
    public void update(Category category) {
        try {
            categoryRepository.save(category);
        }catch (Exception e){
            throw new CategoryNotFoundException("category not found",HttpStatus.NOT_FOUND);
        }

Reply:
it's something similar, in case the save fails, it doesn't mean that the resource was not found, but at least we know there was an error, in case the `save` doesn't return anything it means that it didn't found the resource
i.e
```java
public void update(Category category) {
  try {
    Optional<Category> updatedCategory = categoryRepository.save(category);
    if (!updatedCategory.isPresent()) {
       throw new CategoryNotFoundException("Category not found", HttpStatus.NOT_FOUND);
    }
  }catch (Exception e){
     throw new MoreGeneralException("Error updating category", HttpStatus.SERVER_ERROR);
  }
}
```

> **Note**:
> I didn't test this code, and don't remember everything in Java hehe, try yourself

Reply:
did you see this comments? ☝️

Reply:
In this method, what happens is that a new category is always added, even if it is not a valid id. this example is in Vocabulary Controller but it's pretty much the same thing.
    public void update2(Vocabulary vocabulary) {
        try {
            Optional<Vocabulary> updatedCategory = Optional.of(vocabularyRepository.save(vocabulary));
            if (!updatedCategory.isPresent()) {
                throw new VocabularyNotFoundException("Category not found", HttpStatus.NOT_FOUND);
            }
        }catch (Exception e){
            throw new RuntimeException();
        }
    }

Reply:
> even if it is not a valid id
did you try update a resource with an invalid id using that code? to see what happens?

Reply:
yo intente con UUID no valido y creaba un nuevo objeto, con un id no valido que tampoco era un UUID daba un error ...Dice yendrie que ellos lo hacen asi mas menos como lo primero que habia hecho(que acceden a la BD), ahora mismo no recuerdo si cuando lo probe asi de esa forma me creaba tambien una nuebo objeto.

Reply:
si, vamos a dejarlo asi, y luego le pasamos otra revisada, en general la buena practica es hacer las menos conexiones posibles a DB, pero cuando es necesario es necesario jejhej